lib/active_model/serializer/attribute.rb

```diff
@@ -0,0 +1,38 @@
+module ActiveModel
+  class Serializer
+    Attribute = Struct.new(:name, :block, :key, :condition_type, :condition) do
+      def initialize(name, block, options)
+        super(name, block)
+        self.key = options.fetch(:key, name)
+        self.condition_type =
+          if options.key?(:if)
+            :if
+          elsif options.key?(:unless)
+            :unless
+          else
+            :none
+          end
+        self.condition = options[condition_type]
+      end
+
+      def value(serializer)
+        if block
+          serializer.instance_eval(&block)
+        else
+          serializer.read_attribute_for_serialization(name)
+        end
+      end
+
+      def include?(serializer)
+        case condition_type
+        when :if
+          serializer.send(condition)
+        when :unless
+          !serializer.send(condition)
+        else
+          true
+        end
```

Comment:
Agreed, but I'm not too sure what's best currently. I vote for taking care of it later.